### ISSUE TITLE
VROverlayFlags_SideBySide_Crossed doesn't work properly - possible solution

### ISSUE THREAD

**Issue by Hotrian** [ValveSoftware/openvr#380]
So first of all, I'm using the [Unity SteamVR Plugin](https://www.assetstore.unity3d.com/en/#!/content/32647), so it's totally possible this is due to the C# interface, though I suspect not.

I noticed that `VROverlayFlags_SideBySide_Crossed` and `VROverlayFlags_SideBySide_Parallel` seem to do the same thing. The result is that the left half of the texture is drawn to the left eye, and the right half of the texture is drawn to the right eye.

Incidentally, I noticed that setting `VROverlayFlags_Curved` in addition to either `VROverlayFlags_SideBySide_Crossed` or `VROverlayFlags_SideBySide_Parallel` seems to flip the images as you would expect `VROverlayFlags_SideBySide_Crossed` to do.

I suspect your code is setup similar to this pseudo code:

```C#
void DrawOverlay(Texture tex)
{
	if (VROverlayFlags_SideBySide_Parallel || VROverlayFlags_SideBySide_Crossed)
	{
		Texture[] sbs = SplitTexture(tex);
		if (VROverlayFlags_Curved) // Should be VROverlayFlags_SideBySide_Crossed
		{
			DrawOverlayToLeftEye(sbs[1]);
			DrawOverlayToRightEye(sbs[0]);
		}
		else
		{
			DrawOverlayToLeftEye(sbs[0]);
			DrawOverlayToRightEye(sbs[1]);
		}
	}
	else
	{
		DrawOverlayToLeftEye(tex);
		DrawOverlayToRightEye(tex);
	}
}
```

I tried this using the Unity SteamVR Plugins version 1.1 and 1.2, though as I mentioned I believe the issue lies with SteamVR and not the C# interface, since not much is done there aside from calling the C code, so I did not title this as a Unity issue.

Sorry if this has already been posted. I searched the issues but didn't see anything related.

**Edit:** Also, I am on the latest SteamVR Beta (at least according to Steam), SteamVR Version 2017-01-18 (1484782736). I had issues getting my Vive to work properly after switching back to the main branch, so I would prefer not switching off the beta again if I don't have to. Getting my Vive working again took several restarts and switching between the main and beta branches several times. Would be cool if someone on the main branch could check if `VROverlayFlags_SideBySide_Crossed` works as intended.

**Comment by elvissteinjr** [comment]
Just came across this (non)-behavior regarding the VROverlayFlags_SideBySide_Crossed flag. It still behaves this way in SteamVR Beta 1.8.13, OpenVR API v1.7.15.

VROverlayFlags_Curved has been removed by now, so I couldn't confirm the behavior with that. It's not Unity plugin related, though, as I'm using the C++ API directly.

**Comment by JoeLudwig** [comment]
So, um... I didn't notice this bug when it was filed two and a half years ago. But I saw @elvissteinjr 's note today so I looked at the code. And found what's probably the bug.

The fix will be in 1.8.14. I don't have a test case to verify it, but this was definitely **A** bug, even if it wasn't **the** bug. If it's not fixed, leave a comment here and I'll reopen.
